numba/npyufunc/parfor.py

```diff
@@ -371,7 +371,10 @@ def _create_gufunc_for_parfor_body(
     for var in parfor_redvars:
         arr = var + "_arr"
         parfor_redarrs.append(arr)
-        typemap[arr] = types.npytypes.Array(typemap[var], 1, "C")
+        if arr in typemap:
+            assert(typemap[arr] == types.npytypes.Array(typemap[var], 1, "C"))
```

Comment:
Will this work once the array reduction PR goes in?

Reply:
No.  So, we should wait until that other PR merges then rebase here.  Once that happens, the fix is easy.

Reply:
Ok, great, thanks. Let's wait on that.

Reply:
Done.